xcom2-launcher/xcom2-launcher/Classes/Steam/SteamManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Steamworks;
+
+namespace XCOM2Launcher.Steam
+{
+    public static class SteamManager
+    {
+        private static bool _initialized;
+        private static readonly object _initlock = new object();
+        private static readonly System.Timers.Timer _timer = new System.Timers.Timer(50);
+
+        static SteamManager()
+        {
+            _timer.Elapsed += (sender, args) =>
+            {
+                SteamAPI.RunCallbacks();
+            };
+        }
+        
+        public static bool IsSteamRunning()
+        {
+            return SteamAPI.IsSteamRunning();
+        }
+        
+        public static bool EnsureInitialized()
+        {
+            if (_initialized) return true;
```

Comment:
_initialized is sometimes used inside synchronized block and sometimes used without synchronization.

Reply:
That's called "double-checked locking". There is no need to acquire a lock when _initialized is already true (it cannot become false anymore)